packages/service-integration/src/service_integration/cli/_compose_spec.py

```diff
@@ -34,6 +35,14 @@ def _run_git(*args) -> str:
     ).stdout.strip()
 
 
+def _strip_credentials(url: str) -> str:
+    yarl_url = URL(url)
```

Comment:
- it seems URL("any string here") does not raise anything so it is ok then
- MINOR: might want to keep context for the GC?
```python
if (yarl_url := URL(url)) and yarl_url.is_abloute():
     ... 
return url    
```

Reply:
yes, I had teste that one, also added a ten to make sure it will not fail on us in the future